lib/realtime/server.ex

```diff
@@ -32,57 +29,63 @@ defmodule Realtime.Server do
 
   @doc """
   The subscribing process will get a message when there's new data, with the form
-  {:new_realtime_data, vehicles_by_route_id()}
+  {:new_realtime_data, vehicles_on_route()}
   """
-  @spec subscribe(Route.id(), GenServer.server()) :: vehicles_by_route_id()
+  @spec subscribe(Route.id(), GenServer.server()) :: Vehicles.for_route()
   def subscribe(route_id, server \\ nil) do
     server = server || default_name()
-    {registry_key, vehicles} = GenServer.call(server, {:subscribe, route_id})
+    {registry_key, vehicles_for_route} = GenServer.call(server, {:subscribe, route_id})
     Registry.register(Realtime.Registry, registry_key, route_id)
-    vehicles
+    vehicles_for_route
   end
 
-  @spec update_vehicles(vehicles_by_route_id()) :: term()
-  def update_vehicles(vehicles, server \\ __MODULE__) do
-    GenServer.cast(server, {:update_vehicles, vehicles})
+  @spec update_vehicles(Route.by_id(Vehicles.for_route())) :: term()
+  def update_vehicles(vehicles_by_route_id, server \\ __MODULE__) do
+    GenServer.cast(server, {:update_vehicles, vehicles_by_route_id})
   end
 
   # GenServer callbacks
 
   @impl true
   def init(_opts) do
-    initial_state = %{
-      vehicles: Map.new()
-    }
-
-    {:ok, initial_state}
+    {:ok, initial_state()}
   end
 
+  @spec initial_state() :: state()
+  defp initial_state(), do: %{}
```

Comment:
What's the benefit of breaking this out into a function?

Reply:
Oops, this was left over from a previous change.